cxx4opencl/address_spaces.txt

```diff
@@ -386,14 +387,12 @@ struct C1 {
 
 struct C2 {
   int m;
-  C2(int init) __constant {};
+  constexpr C2(int init) __constant : m(init) {};
 };
 
-kernel void k() {
-  __constant C1 cobj1 = {1};
-  __constant C1 cobj2 = C1();
-  __constant C2 cobj3 = {1}; // error: user defined constructor can't be used
-}
+__constant C1 cobj1 = {1};
+__constant C1 cobj2 = C1();
+__constant C2 cobj3(1);
```

Comment:
Would it make sense to add one negative example where the ctor is invalid which I think was the intent of the original example?

Reply:
The original example doesn't seem to compile due to the clang bug but I think it should also be here to illustrate the restriction?
https://godbolt.org/z/Y53joTYen

Reply:
I have filed a bug llvm.org/PR52437 for this. Once it's fixed we should finalise this PR.

Reply:
This issue seems to be specific to the implementation of OpenCL in Compiler Explorer and not clang. If C++ mode is used the example works correctly:
https://godbolt.org/z/czvYWvK1K